provider/azure/internal/azureauth/interactive.go

```diff
@@ -134,20 +134,45 @@ func InteractiveCreateServicePrincipal(
 		return "", "", errors.Trace(err)
 	}
 
+	return authorizationCreateServicePrincipal(
```

Comment:
Given that a lot of these things are shared with InteractiveCreateServicePrincipal, how about creating an ephemeral type to hold the context, so we can see more easily which things are unique to authorizationCreateServicePrincipal rather than just ambient dependencies?